### PR TITLE
Improve accounts_password_pam_tally2

### PR DESCRIPTION
**Pull request by dodys**
#### Description:

- Adjust rule description
- Add variable var_accounts_passwords_pam_tally2_deny
- Modify OVAL and bash to use variable 
- Add tests

Authored-by: @richardmaciel-canonical

### CODE REVIEW

**Review by jan-cerny — APPROVED** [review]

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/oval/shared.xml [hunk 1]
```diff
@@ -12,13 +12,28 @@
...
     <ind:object object_ref="object_accounts_passwords_pam_tally2_deny_auth" />
+    <ind:state state_ref="state_var_accounts_passwords_pam_tally2_deny_value_upper_bound" />
+    <ind:state state_ref="state_var_accounts_passwords_pam_tally2_deny_value_lower_bound" />
   </ind:textfilecontent54_test>
 
   <ind:textfilecontent54_object id="object_accounts_passwords_pam_tally2_deny_auth" comment="Check deny configuration of pam_tally2" version="1">
     <ind:filepath>/etc/pam.d/common-auth</ind:filepath>
-    <ind:pattern operation="pattern match">^\s*auth(?:(?!\n)\s)+required(?:(?!\n)\s)+pam_tally2.so(?:(?!\n)\s)+(((?:(?!\n)\s)?[^\n]+)?onerr=fail((?:(?!\n)\s)+[^\n]+)?(?:(?!\n)\s)+deny=[123]((\s+\S+)*\s*$))|(((?:(?!\n)\s)?[^\n]+)?deny=[123]((?:(?!\n)\s)+[^\n]+)?(?:(?!\n)\s)+onerr=fail((\s+\S+)*\s*$))</ind:pattern>
+    <ind:pattern operation="pattern match">^\s*auth(?:(?!\n)\s)+required(?:(?!\n)\s)+pam_tally2.so(?:(?!\n)\s)+(?:(?:(?:(?!\n)\s)?[^\n]+)?onerr=fail(?:(?:(?!\n)\s)+[^\n]+)?(?:(?!\n)\s)+deny=([\d]+)(?:(?:\s+\S+)*\s*$))|(?:(?:(?:(?!\n)\s)?[^\n]+)?deny=([\d]+)(?:(?:(?!\n)\s)+[^\n]+)?(?:(?!\n)\s)+onerr=fail(?:(?:\s+\S+)*\s*$))</ind:pattern>
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/oval/shared.xml:21]: The brackets in `[\d]+` are superfluous, you can use just `\d+`.

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/rule.yml [hunk 2]
```diff
@@ -45,16 +45,16 @@ references:
...
 
     <pre># grep pam_tally2.so /etc/pam.d/common-auth
-    auth required pam_tally2.so onerr=fail deny=3</pre>
+    auth required pam_tally2.so onerr=fail deny={{{ xccdf_value("var_accounts_passwords_pam_tally2_deny") }}}</pre>
 
     If no line is returned, the line is commented out, or the line is missing
     "onerr=fail", this is a finding.
-    If the line has "deny" set to a value other than 1, 2, or 3, this is a
-    finding.
+    If the line has "deny" set to a value outside of the range 1-<pre>{{{ xccdf_value("var_accounts_passwords_pam_tally2_deny") }}}</pre>,
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/rule.yml:56]:
```suggestion
    If the line has "deny" set to a value outside of the range <pre>1-{{{ xccdf_value("var_accounts_passwords_pam_tally2_deny") }}}</pre>,
```

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/oval/shared.xml (hunk 1 above)
Comment (dodys) [reply linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/oval/shared.xml]: done, thanks!

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/rule.yml (hunk 2 above)
Comment (dodys) [reply linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_tally2/rule.yml]: fixed, thanks

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**